unified-installer/astra-unified-installer.sh

```diff
@@ -1707,7 +1707,12 @@ step_check_config() {
     # Env vars with special conditions
     if [ -n "$IMAGE_PULL_SECRET" ]; then
         if [ -z "$NAMESPACE" ]; then
-            prompt_user "NAMESPACE" "NAMESPACE is required when specifying an IMAGE_PULL_SECRET. Please enter the namespace:"
+            local -r ns_warning="NAMESPACE is required when specifying an IMAGE_PULL_SECRET"
+            if prompts_disabled; then
+                add_problem "$ns_warning"
+            else
+                prompt_user "NAMESPACE" "$ns_warning. Please enter the namespace:"
+            fi
```

Comment:
With prompts disabled, we were letting the user proceed without a NAMESPACE which would cause a fatal error down the road since we were trying to do operations involving an IMAGE_PULL_SECRET and relied on NAMESPACE also being set.

Reply:
Nice catch!